core/controllers/reader.py

```diff
@@ -310,68 +433,60 @@ def post(self, exploration_id):
         Args:
             exploration_id: str. The ID of the exploration.
         """
-        exp_issue_dict = self.payload.get('exp_issue_dict')
-        try:
-            unused_exp_issue = stats_domain.ExplorationIssue.from_backend_dict(
-                exp_issue_dict)
-        except utils.ValidationError as e:
-            raise self.InvalidInputException(e)
-
         playthrough_data = self.payload.get('playthrough_data')
         try:
             unused_playthrough = stats_domain.Playthrough.from_backend_dict(
                 playthrough_data)
         except utils.ValidationError as e:
             raise self.InvalidInputException(e)
 
+        try:
+            self.current_issue_schema_version = self.payload[
+                'issue_schema_version']
+        except KeyError as e:
+            raise self.InvalidInputException(e)
+
+        try:
+            self.current_playthrough_id = self.payload['playthrough_id']
+        except KeyError as e:
+            raise self.InvalidInputException(e)
+
         exp_version = playthrough_data['exp_version']
 
         exp_issues_model = stats_models.ExplorationIssuesModel.get_model(
             exploration_id, exp_version)
-        exp_issues = stats_services.get_exp_issues_from_model(exp_issues_model)
-
-        customization_args = exp_issue_dict['issue_customization_args']
+        self.current_exp_issues = stats_services.get_exp_issues_from_model(
+            exp_issues_model)
+
+        playthrough = stats_domain.Playthrough.from_dict(playthrough_data)
+
+        # If playthrough already exists, update it in the datastore.
+        if self.current_playthrough_id:
```

Comment:
explicitly check for None: `if self.current_playthrough_id is not None:`

Reply:
Done.